manifests/agent.pp

```diff
@@ -347,9 +347,11 @@
 
   # Controlling the 'zabbix-agent' service
   service { $servicename:
-    ensure  => $service_ensure,
-    enable  => $service_enable,
-    require => $service_require,
+    ensure    => $service_ensure,
+    enable    => $service_enable,
+    require   => $service_require,
+    subscribe => Package[$zabbix_package_agent],
```

Comment:
This should already be present via the `$service_require`, which is defined as:

```
  $service_require = $manage_startup_script ? {
    true  => [Package[$zabbix_package_agent], Zabbix::Startup[$servicename]],
    false => Package[$zabbix_package_agent]
  }
```

If something requires a thing, it will get notified about changes.

Reply:
Do you set `zabbix_package_state => latest`? Otherwise Puppet would not have updated the package and nothing would have issued the notification.